tests/sxg_cert_chain_test.cc

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include "libsxg/sxg_cert_chain.h"
+
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+
+#include <fstream>
+#include <sstream>
+#include <unistd.h>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "test_util.h"
+#include "libsxg/internal/sxg_buffer.h"
+
+namespace {
+
+class CertChainTest : public ::testing::Test {
+ protected:
+  void SetUp() {
+    cert_ = nullptr;
+    issuer_ = nullptr;
+  }
+  void SetEcdsa256() {
+    FILE* const certfile = fopen("testdata/ocsp_included.pem", "r");
+    if (certfile == nullptr) {
+      abort();
+    }
+    cert_ = PEM_read_X509(certfile, nullptr, nullptr, nullptr);
+    issuer_ = PEM_read_X509(certfile, nullptr, nullptr, nullptr);
+    fclose(certfile);
+  };
+  void SetBadCert() {
+    FILE* const certfile = fopen("testdata/cert256.pem", "r");
+    if (certfile == nullptr) {
+      abort();
+    }
+    cert_ = PEM_read_X509(certfile, nullptr, nullptr, nullptr);
+    issuer_ = PEM_read_X509(certfile, nullptr, nullptr, nullptr);
+    fclose(certfile);
+  };
+  void TearDown() {
+    if (cert_) {
+      X509_free(cert_);
+    }
+    if (issuer_) {
+      X509_free(issuer_);
+    }
+  }
+  X509* cert_;
+  X509* issuer_;
+};
+
+TEST_F(CertChainTest, ExtractOcspUri) {
+  SetEcdsa256();
+  sxg_buffer_t url = sxg_empty_buffer();
+
+  EXPECT_TRUE(sxg_extract_ocsp_url(cert_, &url));
+  EXPECT_EQ('\0', url.data[url.size - 1]);  // Expects null termination.
+  url.size -= 1;  // remove the null termination.
+  ASSERT_EQ("http://ocsp.digicert.com", sxg_test::BufferToString(url));
+
+  sxg_buffer_release(&url);
+}
+
+TEST_F(CertChainTest, SendRequest) {
+  int fds[2];
+  ASSERT_EQ(0, pipe(fds));
+  BIO *mem = BIO_new_fd(fds[1], /*close_flag*/1);
+  SetEcdsa256();
+  OCSP_RESPONSE* result = nullptr;
+  std::string buff;
+  buff.resize(4096);
+  memset(&buff[0], 0, 4096);
```

Comment:
Does `std::string buff(4096, '\0')` express this more naturally?

Reply:
I agree. done.